terraspace/app/stacks/peer/main.tf

```diff
@@ -0,0 +1,172 @@
+  public_subnets       = ["10.0.5.0/24", "10.0.6.0/24", "10.0.7.0/24"]                # LoadBalancer and NAT
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+  enable_dns_hostnames = true
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                    = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"           = "1"
+  }
+}
+
+module "gateway_endpoint_to_dynamodb" {
+  source         = "../../modules/gateway-endpoint-to-dynamodb"
+  vpc_id         = module.vpc.vpc_id
+  region         = var.region
+  route_table_id = module.vpc.private_route_table_ids[0]
+}
+
+module "gateway_endpoint_to_s3" {
+  source         = "../../modules/gateway-endpoint-to-s3"
+  vpc_id         = module.vpc.vpc_id
+  region         = var.region
+  route_table_id = module.vpc.private_route_table_ids[0]
+}
+
+data "http" "myip" {
+  url = "http://ipv4.icanhazip.com"
+}
+
+module "eks" {
+  source                             = "terraform-aws-modules/eks/aws"
+  version                            = "~> 18.2.0"
+  cluster_name                       = var.cluster_name
+  cluster_version                    = var.cluster_version
+  cluster_endpoint_private_access    = true
+  cluster_endpoint_public_access     = true
+  vpc_id                             = module.vpc.vpc_id
+  subnet_ids                         = [module.vpc.private_subnets[0], module.vpc.private_subnets[1]]
+  enable_irsa                        = true # To be able to access AWS services from PODs  
+  cluster_security_group_description = "EKS cluster security group - Control Plane"
+
+  cluster_endpoint_public_access_cidrs = [ # TODO: Access through AWS transit gateway
+    "${chomp(data.http.myip.body)}/32",    # GitHub Actions Self Runner Static IP 
+    "177.33.141.81/32",
+    "185.152.47.29/32",
+    "168.227.34.17/32",
+  ]
+
+  eks_managed_node_groups = { # Needed for CoreDNS (https://docs.aws.amazon.com/eks/latest/userguide/fargate-getting-started.html)
+    test-ipfs-peer-subsys = {
+      name         = var.cluster_name
+      desired_size = 2
+      min_size     = 2
+      max_size     = 20
+
+      instance_types = ["c6i.2xlarge"]
+      k8s_labels = {
+        workerType = "managed_ec2_node_groups"
+      }
+      update_config = {
+        max_unavailable_percentage = 50
+      }
+
+      tags = { # This is also applied to IAM role.
+        "eks/${var.account_id}/${var.cluster_name}/type" : "node"
+        "k8s.io/cluster-autoscaler/${var.cluster_name}" : "owned"
+        "k8s.io/cluster-autoscaler/enabled" : "TRUE"
+      }
+    }
+  }
+
+  node_security_group_additional_rules = local.node_security_group_additional_rules
+
+  fargate_profiles = {
+    default = {
+      name       = "default"
+      subnet_ids = [module.vpc.private_subnets[2], module.vpc.private_subnets[3]]
```

Comment:
is there a reason we are only utilizing fargate in these specific availability zones?